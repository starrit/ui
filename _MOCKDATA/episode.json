{
  "uid": "5F59D81F-C089-2397-8CD7-247C1E23B191",
  "url": "https://podcast.ustudio.com/show/5F59D81F-C089-2397-8CD7-247C1E23B191",
  "title": "Lorem ipsum dolor",
  "description": "Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Curabitur sed tortor. Integer aliquam adipiscing lacus. Ut nec urna et arcu imperdiet ullamcorper. Duis at lacus. Quisque purus sapien, gravida non, sollicitudin a, malesuada id, erat. Etiam vestibulum massa rutrum magna. Cras convallis convallis dolor. Quisque tincidunt pede ac urna. Ut tincidunt vehicula",
  "show_uid": "0CCBBCD9-1BE5-471B-2738-B5DB1DC2B642",
  "image_url": "/assets/media/show_thumb_2.png",
  "show_url": "/show/0CCBBCD9-1BE5-471B-2738-B5DB1DC2B642",
  "published_at": "1510014877",
  "duration": 42537063,
  "source": {
    "filename": "IMG_2116.MOV",
    "src": "https://podcast.ustudio.com/media/IMG_2116.MOV",
    "size": 130023424,
    "published_at": 1525269921660
  }
}
