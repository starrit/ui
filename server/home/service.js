import React from 'react';
import { renderToString } from 'react-dom/server';
import template from '../template';

module.exports = class CareService {

  /**
   * Gets Promise of PreRendered Care Page.
   * Returns null if current etag would match etag passed in.
   * Error is thrown if data for desired region/locality/type is not present.
   * @param existingEtag passed in etag for comparison.
   * @param type type of care centers desired.
   * @param region region of care centers desired.
   * @param locality locality of care centers desired.
   * @param page current page requested.
   * @param path path of current request.
   * @returns {Promise.<T>}
   */
  getHomePage(existingEtag, type, region, locality, page, path) {
    // GET Care data from API.
    React.createElement()
    const appString = renderToString(<h1>Hello</h1>)
    const out = {
      body : template({
        body: appString,
        title: 'StarrIT L.L.C. Providing a complete set of consulting services to meet your business needs. ',
        initialState: JSON.stringify({}),
        cssBundle: '', //ENV['main.css'],
        jsBundle: 'dist/main.js'
      })};

    return out;
  }

  /**
   * Validates data returned from API to ensure that data for given locality exists.
   * @param data data returned from API.
   * @param care_type type of desired care centers.
   * @param region region of desired care centers.
   * @param locality of desired care centers.
   */
  validateCareData(data, care_type, region, locality)  {

    // ensure that 'meta' section of response contains at least one item including locality information.
    const meta = data['meta'];
    let locality_info;
    for (var i in meta) {
      if (meta[i]['locality']) {
        locality_info = meta[i]['locality'];
        break;
      }
    }
    if (!locality_info) {
      throw new Error(`Locality info not found for: ${care_type}/${region}/${locality}`)
    }

    // ensure that the type of at least one included section is 'destination_page'
    const included = data['included'];
    let cms_content;
    for (var j in included) {
      if (included[j]['type'] === 'destination_page') {
        cms_content = true;
        break;
      }
    }
    if (!cms_content) {
      throw new Error(`CMS content not found for: ${care_type}/${region}/${locality}`)
    }
  }

  postCareService(type, region, locality) {
    return Promise.resolve();
  }
};
