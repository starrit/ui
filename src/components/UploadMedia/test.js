import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import xhrMock from 'xhr-mock';

configure({adapter: new Adapter()});

import appConstants from 'constants/app-constants';
import UploadMediaConnected, {UploadMedia} from './';
import * as episodeService from 'AppServices/episode.service';
jest.mock('AppServices/episode.service');

const mockAcceptedFiles = [
  {
    name: 'Mock File 1.mp4',
    size: 400 * 1024 * 1024,
    loaded: 40,
    preview: 42
  },
  {
    name: 'Mock File 2.mp4',
    size: 900 * 1024 * 1024,
    loaded: 17,
    preview: 58
  }
];

const mockRejectedFiles = [
  {
    name: 'unsupported-file-type.txt',
    size: 40 * 1024,
    preview: 87
  },
  {
    name: 'very-large-file.mp4',
    size: appConstants.MAX_MEDIA_SIZE * 1.5,
    preview: 93
  }
];

describe('Upload Media Component', () => {
  beforeAll(() => {
    episodeService.uploadProgress = () => {
      return Promise.resolve({state: 'uploading', size: 1, received: 1});
    }
    episodeService.uploadMedia = (file, progressFunction, completeFunction) => {
      return Promise.resolve({});
    };
  });
  it('should render the component', () => {
    const wrapper = shallow(<UploadMedia />);
    wrapper.setState({
      files: mockAcceptedFiles
    });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should handle file drop', done => {
    const addNotificationMock = jest.fn();
    const wrapper = mount(<UploadMedia isModalOpen={true} addNotification={addNotificationMock} />);

    wrapper.instance().onDrop(mockAcceptedFiles, mockRejectedFiles);

    // Rejected Files notification
    expect(addNotificationMock).toHaveBeenCalledTimes(2);
    setTimeout(() => {
      expect(addNotificationMock).toHaveBeenCalledTimes(2);
      done();
    }, 100);
  });

  it('should redirect the user to Media Edit page on clicking notification', () => {
    const wrapper = mount(<UploadMedia history={[]} />);
    wrapper.instance().goToMediaEdit({id: 5});
    expect(wrapper.prop('history')).toEqual(['/media/5/edit']);
  });
});
