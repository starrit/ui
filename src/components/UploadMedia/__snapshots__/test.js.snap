// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Media Component should render the component 1`] = `
<div
  className="backdrop hidden"
>
  <div
    className="modalWrap"
  >
    <header
      className="modalHeader"
    >
      <H3
        className=""
      >
        Upload Media
      </H3>
      <a>
        <Icon
          className=""
          viewBoxHeight={50}
          viewBoxWidth={50}
        >
          <path
            d="M27.3,24.9L48.1,4.1c0.7-0.7,0.7-1.9,0-2.6c-0.7-0.7-1.9-0.7-2.6,0L24.8,22.3L4.1,1.7c-0.7-0.7-1.9-0.7-2.6,0s-0.7,1.9,0,2.6l20.7,20.7L1.7,45.5c-0.7,0.7-0.7,1.9,0,2.6c0.4,0.4,0.8,0.5,1.3,0.5c0.5,0,0.9-0.2,1.3-0.5l20.6-20.6l20.7,20.7c0.4,0.4,0.8,0.5,1.3,0.5c0.5,0,0.9-0.2,1.3-0.5c0.7-0.7,0.7-1.9,0-2.6L27.3,24.9z"
          />
        </Icon>
      </a>
    </header>
    <div
      className="modalBody"
    >
      <t
        accept="video/*"
        activeClassName="dropzoneActive"
        className="dropzone"
        disableClick={false}
        disablePreview={false}
        disabled={false}
        maxSize={4294967296}
        minSize={0}
        multiple={true}
        onDrop={[Function]}
        preventDropOnDocument={true}
      >
        <H3
          className=""
        >
          Drag and drop file here
        </H3>
      </t>
      <div
        className="uploadFilesWrap"
      >
        <FlexContainer
          alignContent="flex-start"
          alignItems="center"
          className="uploadingFile"
          direction="row"
          justifyContent="flex-start"
          key="42"
          widthPercent={100}
          wrap="nowrap"
        >
          <P
            className=""
          >
            <span
              title="Mock File 1.mp4"
            >
              Filename: 
              Mock File 1.mp4
            </span>
          </P>
          <P
            className=""
          >
            <span>
              File Size: 
              400 MB
            </span>
          </P>
          <ProgressBar
            circular={false}
            loaded={40}
          />
        </FlexContainer>
        <FlexContainer
          alignContent="flex-start"
          alignItems="center"
          className="uploadingFile"
          direction="row"
          justifyContent="flex-start"
          key="58"
          widthPercent={100}
          wrap="nowrap"
        >
          <P
            className=""
          >
            <span
              title="Mock File 2.mp4"
            >
              Filename: 
              Mock File 2.mp4
            </span>
          </P>
          <P
            className=""
          >
            <span>
              File Size: 
              900 MB
            </span>
          </P>
          <ProgressBar
            circular={false}
            loaded={17}
          />
        </FlexContainer>
      </div>
    </div>
  </div>
</div>
`;
